### PR TITLE
feat: New password cannot be same as old password

### PR DESCRIPTION
**Pull request by laurensmertens**
|Q              |A  |
|---------------|---|
|Bug fix?       |No|
|New feature?   |Yes|	
|BC breaks?     |No|	
|Deprecations?  |No|	
|Fixed tickets  |No|	

Validation to make sure users choose a new password when resetting.

### CODE REVIEW

File: Resources/config/validation.xml [hunk 1]
```diff
@@ -0,0 +1,100 @@
+<?xml version="1.0" ?>
+<constraint-mapping xmlns="http://symfony.com/schema/dic/constraint-mapping"
+    xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+    xsi:schemaLocation="http://symfony.com/schema/dic/constraint-mapping
+        http://symfony.com/schema/dic/constraint-mapping/constraint-mapping-1.0.xsd">
+
+    <class name="FOS\UserBundle\Model\User">
+
+        <constraint name="EMS\CoreBundle\Validator\Constraints\IsDifferentPassword">
+            <option name="message">Password has to be different from the previous onesssss</option>
```
Comment (Davidmattei) [inline Resources/config/validation.xml:10]: sssss ?

File: Validator/Constraints/IsDifferentPasswordValidator.php [hunk 2]
```diff
@@ -0,0 +1,23 @@
+<?php
+
+namespace EMS\CoreBundle\Validator\Constraints;
+
+use Symfony\Component\Validator\Constraint;
+use Symfony\Component\Validator\ConstraintValidator;
+
+class IsDifferentPasswordValidator extends ConstraintValidator
+{
+    const ERROR_MESSAGE = 'Password has to be different from the previous one';
```
Comment (Davidmattei) [inline Validator/Constraints/IsDifferentPasswordValidator.php:10]: use the message from the constraint ?

**Review by Davidmattei — CHANGES_REQUESTED** [review]

File: Validator/Constraints/IsDifferentPasswordValidator.php (hunk 2 above)
Comment (laurensmertens) [reply Validator/Constraints/IsDifferentPasswordValidator.php:10]: PHPStan does not allow this, I tried $constraint->message but I get the error "Access to an undefined property Symfony\Component\Validator\Constraint::$message." Although the code works.

File: Resources/config/validation.xml (hunk 1 above)
Comment (laurensmertens) [reply Resources/config/validation.xml]: Fixed :-)

**Comment by dameert** [comment]
Resync with develop

File: Twig/AppExtension.php [hunk 3]
```diff
@@ -102,27 +102,27 @@ public function __construct(Registry $doctrine, AuthorizationCheckerInterface $a
     public function getFunctions()
     {
         return [
```
Comment (Davidmattei) [inline Twig/AppExtension.php:104]: this is not good, please remove from PR

**Review by Davidmattei — CHANGES_REQUESTED** [review]
Please cleanup commit, maybe do a rebase?

**Pull request closed without merging**